lib/Runtime/Library/JavascriptPromise.cpp

```diff
@@ -201,16 +201,18 @@ namespace Js
         remainingElementsWrapper->remainingElements = 1;
 
         JavascriptExceptionObject* exception = nullptr;
-
+        RecyclableObject* iterator = nullptr;
+        bool shouldCallReturn = false;
         try
         {
             // 4. Let iterator be GetIterator(iterable).
-            RecyclableObject* iterator = JavascriptOperators::GetIterator(iterable, scriptContext);
+            iterator = JavascriptOperators::GetIterator(iterable, scriptContext);
             Var next;
             values = library->CreateArray(0);
 
             while (JavascriptOperators::IteratorStepAndValue(iterator, scriptContext, &next))
```

Comment:
Is there a reason not to use `JavascriptOperators::DoIteratorStepAndValue` here instead of doing it manually?  This feels error prone.  Same with `EntryRace` below.